Packs/Illumio/Integrations/IllumioCore/IllumioCore.py

```diff
@@ -0,0 +1,1232 @@
+            break
+    else:
+        rule = Rule.build(
+            ingress_services=ingress_services,
+            consumers=consumers, providers=providers,
+            resolve_consumers_as=resolve_consumers_as,
+            resolve_providers_as=resolve_providers_as
+        )
+        response = client.rules.create(rule, parent=ruleset_href)
+
+    response = response.to_json()
+    readable_output = prepare_rule_create_output(response)  # type: ignore
+
+    return CommandResults(
+        outputs_prefix="Illumio.Rule",
+        outputs_key_field="href",
+        outputs=remove_empty_elements(response),
+        readable_output=readable_output,
+        raw_response=response
+    )
+
+
+def main():
+    """Parse params and runs command functions."""
+    try:
+        command = demisto.command()
+
+        params = demisto.params()
+        api_user = params.get("api_user")
+        api_key = params.get("api_key")
+
+        port = arg_to_number(params.get("port"), required=True, arg_name="port")
+        if port < MIN_PORT or port > MAX_PORT:  # type: ignore
+            raise InvalidValueError(
+                message="{} is an invalid value for port. Value must be in 1 to 65535.".format(port))
+
+        org_id = arg_to_number(params.get("org_id"), required=True, arg_name="org_id")
+        if org_id <= 0:  # type: ignore
+            raise ValueError(
+                "{} is an invalid value. Organization ID must be a non-zero and positive numeric value.".format(
+                    org_id
+                )
+            )
+
+        base_url = params.get("url").strip()
+        if not base_url:
+            raise ValueError("Server URL is required.")
+
+        proxy = handle_proxy()
+
+        client = PolicyComputeEngine(url=base_url, port=port, org_id=org_id)
+        client.set_proxies(
+            http_proxy=proxy.get("http", None), https_proxy=proxy.get("https", None)
+        )
+        client.set_credentials(api_user, api_key)
+
+        if command == "test-module":
+            return_results(test_module(client))
+        else:
+            illumio_commands = {
+                "illumio-traffic-analysis": traffic_analysis_command,
+                "illumio-virtual-service-create": virtual_service_create_command,
+                "illumio-service-binding-create": service_binding_create_command,
+                "illumio-object-provision": object_provision_command,
+                "illumio-workload-get": workload_get_command,
+                "illumio-workloads-list": workloads_list_command,
+                "illumio-enforcement-boundary-create": enforcement_boundary_create_command,
+                "illumio-enforcement-mode-update": update_enforcement_mode_command,
+                "illumio-ip-list-get": ip_list_get_command,
+                "illumio-ip-lists-get": ip_lists_get_command,
+                "illumio-ruleset-create": ruleset_create_command,
+                "illumio-rule-create": rule_create_command,
```

Comment:
what is the difference between this 2 commands?

Reply:
Ruleset is the superset where all the rules will be linked, where the rule is specific with more details. All rules should be linked to any ruleset.